OpenRA.Mods.RA2/Traits/SpawnNeighboringActors.cs

```diff
@@ -0,0 +1,86 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	[Desc("This actor places other actors around itself.")]
+	public class SpawnNeighboringActorsInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[ActorReference]
+		[Desc("Types of actors to place. If multiple are defined, a random one will be selected.")]
+		public readonly HashSet<string> ActorTypes = new HashSet<string>();
+
+		[FieldLoader.Require]
+		[Desc("Locations to place the mines, from top-left of the building.")]
+		public readonly CVec[] Locations = { };
+
+		public object Create(ActorInitializer init) { return new SpawnNeighboringActors(this, init.Self); }
+	}
+
+	public class SpawnNeighboringActors : INotifyKilled, INotifyOwnerChanged, INotifyActorDisposing
+	{
+		readonly SpawnNeighboringActorsInfo info;
+		readonly List<Actor> actors = new List<Actor>();
+
+		public SpawnNeighboringActors(SpawnNeighboringActorsInfo info, Actor self)
+		{
+			this.info = info;
+
+			SpawnActors(self);
+		}
+
+		public void SpawnActors(Actor self)
+		{
+			foreach (var offset in info.Locations)
+			{
+				var cell = self.Location + offset;
+				var actor = info.ActorTypes.Random(self.World.SharedRandom).ToLowerInvariant();
+				{
```

Comment:
Why are we creating another scope here?
Should this be a frame-end task?

Reply:
Yes, that was actually broken probably from an incomplete copy and paste. Fixed.